Gems/Atom/Feature/Common/Code/Source/PostProcessing/HDRColorGradingPass.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+ 
+#include <PostProcessing/HDRColorGradingPass.h>
+#include <PostProcess/PostProcessFeatureProcessor.h>
+
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/View.h>
+
+ namespace AZ
+{
+    namespace Render
+    {
+        RPI::Ptr<HDRColorGradingPass> HDRColorGradingPass::Create(const RPI::PassDescriptor& descriptor)
+        {
+            RPI::Ptr<HDRColorGradingPass> pass = aznew HDRColorGradingPass(descriptor);
+            return AZStd::move(pass);
+        }
+        
+        HDRColorGradingPass::HDRColorGradingPass(const RPI::PassDescriptor& descriptor)
+            : AZ::RPI::FullscreenTrianglePass(descriptor)
+        {
+        }
+        
+        void HDRColorGradingPass::InitializeInternal()
+        {
+            FullscreenTrianglePass::InitializeInternal();
+
+            m_colorGradingExposureIndex.Reset();
+            m_colorGradingContrastIndex.Reset();
+            m_colorGradingHueShiftIndex.Reset();
+            m_colorGradingPreSaturationIndex.Reset();
+            m_colorFilterIntensityIndex.Reset();
+            m_colorFilterMultiplyIndex.Reset();
+            m_whiteBalanceKelvinIndex.Reset();
+            m_whiteBalanceTintIndex.Reset();
+            m_splitToneBalanceIndex.Reset();
+            m_splitToneWeightIndex.Reset();
+            m_colorGradingPostSaturationIndex.Reset();
+            m_smhShadowsStartIndex.Reset();
+            m_smhShadowsEndIndex.Reset();
+            m_smhHighlightsStartIndex.Reset();
+            m_smhHighlightsEndIndex.Reset();
+            m_smhWeightIndex.Reset();
+
+            m_channelMixingRedIndex.Reset();
+            m_channelMixingGreenIndex.Reset();
+            m_channelMixingBlueIndex.Reset();
+
+            m_colorFilterSwatchIndex.Reset();
+            m_splitToneShadowsColorIndex.Reset();
+            m_splitToneHighlightsColorIndex.Reset();
+            m_smhShadowsColorIndex.Reset();
+            m_smhMidtonesColorIndex.Reset();
+            m_smhHighlightsColorIndex.Reset();
+        }
+        
+        void HDRColorGradingPass::FrameBeginInternal(FramePrepareParams params)
+        {
+            SetSrgConstants();
+
+            FullscreenTrianglePass::FrameBeginInternal(params);
+        }
+
+        bool HDRColorGradingPass::IsEnabled() const
+        {
+            auto* colorGradingSettings = GetHDRColorGradingSettings();
```

Comment:
Is it possible to make this a const pointer?

Reply:
Fixed!